shell/platform/glfw/text_input_plugin.cc

```diff
@@ -42,7 +42,7 @@ void TextInputPlugin::CharHook(GLFWwindow* window, unsigned int code_point) {
   }
   // TODO(awdavies): Actually handle potential unicode characters. Probably
   // requires some ICU data or something.
-  active_model_->AddCharacter(static_cast<char>(code_point));
+  active_model_->AddCharacter(code_point);
```

Comment:
Per https://www.glfw.org/docs/latest/input_guide.html#input_char the code_point here is UTF-32. That means we need to do a UTF-32->UTF-16 conversion, and then call `AddCharacter` one or more times depending on the resulting length, right?

Reply:
We use utf32 as the internal encoding, which ensures that there is no 2-4 byte inconsistency, and the correctness can be guaranteed in code-dependent offsets such as find/insert.

Reply:
I was thinking about this more last night, and it's not clear to me that using even UTF-32 internally is particularly useful in the long term. It's definitely going to be less broken than the old code with minimal changes, but it's still not going to correctly handling combining characters. We'll need real unicode handling at some point to make this code correct, at which point I don't think UTF-32 will be more useful that UTF-8, and since the transport layer is using UTF-8 using UTF-8 would avoid converting the whole string back and forth all the time.

We can keep AddCharacter using UTF-32 if it's the most convenient option for the callers, but we shouldn't decide based on what the internal representation is currently using.